kubernetes/nodecollection.go

```diff
@@ -136,7 +137,9 @@ func downloadNodeData(ctx context.Context, prefix string, config KubeAgentConfig
 			}
 			err := retrieveNodeData(nd, config, nodeSource, currentNode)
 			if err != nil {
+				m.Lock()
```

Comment:
Need a lock here too: https://github.com/cloudability/metrics-agent/pull/194/files#diff-7c1ca396f7b3ef69463ad472cf0120e4a854574ec4d837eea82655a8742a300dR127

Ideally would add tests for this as well. Can make a follow story for that if you want though.

Reply:
Little confused, the comment says need a lock here too, and it is on the m.Lock() line for me.

Reply:
Weird, the link isn't working. On line 127 there needs to be a lock around this line:
```
			failedNodeList[currentNode.Name] = errors.New("provider ID for node does not exist. " +
"If this condition persists it will cause inconsistent cluster allocation")
```

Reply:
thanks for clarifying, added the lock